packages/plugin-http-client/src/Processor.ts

```diff
@@ -0,0 +1,28 @@
+import { HttpAgentResponse, Dependencies } from '@ima/core';
+
+import { HttpClientRequest } from './HttpClient';
+
+export enum Operation {
+  PRE_REQUEST = 'preRequest',
+  POST_REQUEST = 'postRequest',
+}
+
+export type ProcessorParams<B> = {
+  response: HttpAgentResponse<B> | null;
+  request: HttpClientRequest;
+  additionalParams?: any;
+};
+
+export class Processor {
```

Comment:
Same as `Mapper` I woudl drop the base class in favor of an interface (that only describes the type of pre and post request methods)

Reply:
Added interface for Processor and changed to abstract AbstractProcessor.